### PR TITLE
change GlobalScope by using delegation MainScope

### PR DESCRIPTION
**Pull request by underwindfall**
We should not use GlobalScope it's not recommanded by Kotlin.
You can see more details in this [article written by Roman Elizarov](https://medium.com/@elizarov/the-reason-to-avoid-globalscope-835337445abc)

### CODE REVIEW

File: androidApp/src/main/java/xyz/mlumeau/kosmos/views/MainActivity.kt [hunk 1]
```diff
@@ -39,7 +36,7 @@ class MainActivity : AppCompatActivity() {
     }
 
     private fun getAPOD() {
-        GlobalScope.launch {
+        launch {
             apodRepository.getAPOD()?.let { apod ->
                 withContext(Dispatchers.Main) {
```
Comment (RobinCaroff) [inline androidApp/src/main/java/xyz/mlumeau/kosmos/views/MainActivity.kt]: If we use the MainScope then switching to the Main thread becomes redundant. Please remove the `withContext(Dispatchers.Main) {}` and we're good to go.

**Comment by RobinCaroff** [comment]
@underwindfall Thanks for  your feedback and the PR !  I suggested one changed before approving it.
Just so you know: iIn a close future we may remove the coroutines from the Android app as it seems to overcomplicate the workshop.

File: androidApp/src/main/java/xyz/mlumeau/kosmos/views/MainActivity.kt [hunk 2]
```diff
@@ -39,7 +36,7 @@ class MainActivity : AppCompatActivity() {
     }
 
     private fun getAPOD() {
-        GlobalScope.launch {
+        launch {
             apodRepository.getAPOD()?.let { apod ->
```
Comment (RobinCaroff) [inline androidApp/src/main/java/xyz/mlumeau/kosmos/views/MainActivity.kt]: As we use the MainScope, the Main thread dispatcher will be used by default. You should wrap the `getAPOD()` call with a `withContext(Dispatchers.IO)` instruction to perform the deserialisation on a background thread.

Comment (underwindfall) [reply androidApp/src/main/java/xyz/mlumeau/kosmos/views/MainActivity.kt]: or we can just do something like this 
```kotlin
launch(Dispatchers.IO){
}
```

**Comment by underwindfall** [comment]
The PR has been updated you can recheck it .
@mlumeau

**Comment by RobinCaroff** [comment]
Perfect! Thanks!

**Pull request merged**